subworkflows/nf-core/bam_markduplicates_picard/main.nf

```diff
@@ -46,3 +46,21 @@ workflow BAM_MARKDUPLICATES_PICARD {
 
     versions = ch_versions                       // channel: [ versions.yml ]
 }
+
+workflow {
+    main:
+    def ch_bam = Channel.empty()
+    def ch_fasta = Channel.empty()
+    def ch_fai = Channel.empty()
+
+    BAM_MARKDUPLICATES_PICARD( ch_bam, ch_fasta, ch_fai )
+
+    publish:
+    BAM_MARKDUPLICATES_PICARD.out.bam       >> 'picard'
+    BAM_MARKDUPLICATES_PICARD.out.metrics   >> 'picard/metrics'
+    BAM_MARKDUPLICATES_PICARD.out.bai       >> 'picard'
+    BAM_MARKDUPLICATES_PICARD.out.csi       >> 'picard'
+    BAM_MARKDUPLICATES_PICARD.out.stats     >> 'picard/samtools_stats'
+    BAM_MARKDUPLICATES_PICARD.out.flagstat  >> 'picard/samtools_stats'
+    BAM_MARKDUPLICATES_PICARD.out.idxstats  >> 'picard/samtools_stats'
+}
```

Comment:
Here's an example of how to provide example publish targets for a module/subworkflow while keeping them in an entry workflow. This way the module/subworkflow can be invoked directly (similar to nf-wrap) and could even supplement/replace the nf-test DSL

Reply:
I've been heavily in favour of this since I first heard it. Every component becomes a runnable thing.